jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -103,11 +103,35 @@
 
     private enum EncodingFrom
     {
-        NOT_SET, INFERRED, SET_LOCALE, SET_CONTENT_TYPE, SET_CHARACTER_ENCODING
+        /**
+         * Character encoding was not set, or the encoding was cleared with {@code setCharacterEncoding(null)}.
+         */
+        NOT_SET,
+
+        /**
+         * Character encoding was not explicitly set but has a default value defined by the {@code encoding.properties} file.
+         * @see MimeTypes#getInferredEncodings().
+         */
```

Comment:
is that the only mechanism to get an inferred type?
I think it would be best just to say that  the charset is inferred from the content-type and will be added as a parameter to the content-type